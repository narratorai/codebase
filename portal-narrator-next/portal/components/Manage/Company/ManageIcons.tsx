import {
  ApiOutlined,
  DollarOutlined,
  FileSyncOutlined,
  HddOutlined,
  ProfileOutlined,
  TeamOutlined,
} from '@ant-design/icons'

import { ManagePaths } from './interfaces'

interface Props {
  type: ManagePaths
}

const ManageIcons = ({ type, ...iconProps }: Props) => {
  if (type === 'warehouse') {
    return <HddOutlined {...iconProps} />
  }

  if (type === 'company') {
    return <ProfileOutlined {...iconProps} />
  }

  if (type === 'users') {
    return <TeamOutlined {...iconProps} />
  }

  if (type === 'billing') {
    return <DollarOutlined {...iconProps} />
  }

  if (type === 'branding') {
    return <FileSyncOutlined {...iconProps} />
  }

  if (type === 'connections') {
    return <ApiOutlined {...iconProps} />
  }

  return null
}

export default ManageIcons
