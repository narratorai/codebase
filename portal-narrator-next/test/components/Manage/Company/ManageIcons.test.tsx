import { render } from '@testing-library/react'
import ManageIcons from '../../../../portal/components/Manage/Company/ManageIcons'

test('warehouse type returns correct icon', () => {
  const { asFragment } = render(<ManageIcons type="warehouse" />)
  expect(asFragment()).toMatchSnapshot()
})

test('company type returns correct icon', () => {
  const { asFragment } = render(<ManageIcons type="company" />)
  expect(asFragment()).toMatchSnapshot()
})

test('users type returns correct icon', () => {
  const { asFragment } = render(<ManageIcons type="users" />)
  expect(asFragment()).toMatchSnapshot()
})

test('billing type returns correct icon', () => {
  const { asFragment } = render(<ManageIcons type="billing" />)
  expect(asFragment()).toMatchSnapshot()
})
