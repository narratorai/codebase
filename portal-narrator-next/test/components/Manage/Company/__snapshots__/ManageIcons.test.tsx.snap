// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`billing type returns correct icon 1`] = `
<DocumentFragment>
  <span
    aria-label="dollar"
    class="anticon anticon-dollar"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="dollar"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`company type returns correct icon 1`] = `
<DocumentFragment>
  <span
    aria-label="profile"
    class="anticon anticon-profile"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="profile"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656zM492 400h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zM340 368a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`users type returns correct icon 1`] = `
<DocumentFragment>
  <span
    aria-label="team"
    class="anticon anticon-team"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="team"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M824.2 699.9a301.55 301.55 0 00-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 00-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 008 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 01612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 008-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 01-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 01612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 01-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 008 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`warehouse type returns correct icon 1`] = `
<DocumentFragment>
  <span
    aria-label="hdd"
    class="anticon anticon-hdd"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="hdd"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-600 72h560v208H232V136zm560 480H232V408h560v208zm0 272H232V680h560v208zM496 208H312c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM312 544h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H312c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm328 244a40 40 0 1080 0 40 40 0 10-80 0z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
